components/edit-collective/EditVirtualCardModal.js

```diff
@@ -0,0 +1,261 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useLazyQuery, useMutation } from '@apollo/client';
+import { useFormik } from 'formik';
+import { debounce } from 'lodash';
+import { FormattedMessage } from 'react-intl';
+
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+
+import CollectivePicker from '../CollectivePicker';
+import Container from '../Container';
+import { Grid } from '../Grid';
+import StyledButton from '../StyledButton';
+import StyledHr from '../StyledHr';
+import StyledInput from '../StyledInput';
+import StyledInputAmount from '../StyledInputAmount';
+import StyledInputField from '../StyledInputField';
+import Modal, { ModalBody, ModalFooter, ModalHeader } from '../StyledModal';
+import { P } from '../Text';
+import { TOAST_TYPE, useToasts } from '../ToastProvider';
+
+const editVirtualCardMutation = gqlV2/* GraphQL */ `
+  mutation editVirtualCard(
+    $virtualCard: VirtualCardReferenceInput!
+    $name: String!
+    $monthlyLimit: AmountInput
+    $assignee: AccountReferenceInput!
+  ) {
+    editVirtualCard(virtualCard: $virtualCard, name: $name, monthlyLimit: $monthlyLimit, assignee: $assignee) {
+      id
```

Comment:
A good practice here is usually to fetch the properties you just updated, so Apollo can update them in cache.